sdk/messaging/azservicebus/sender.go

```diff
@@ -93,9 +95,18 @@ type SendMessageBatchOptions struct {
 // Message batches can be created using [Sender.NewMessageBatch].
 // If the operation fails it can return an [*azservicebus.Error] type if the failure is actionable.
 func (s *Sender) SendMessageBatch(ctx context.Context, batch *MessageBatch, options *SendMessageBatchOptions) error {
+	to := &tracing.TracerOptions{
+		Tracer:   s.tracer,
+		SpanName: tracing.SendBatchSpanName,
+		Attributes: append(
+			getSenderSpanAttributes(s.queueOrTopic, tracing.SendOperationName),
+			getMessageBatchSpanAttributes(int(batch.NumMessages()))...,
+		),
+	}
+
 	err := s.links.Retry(ctx, EventSender, "SendMessageBatch", func(ctx context.Context, lwid *internal.LinksWithID, args *utils.RetryFnArgs) error {
 		return lwid.Sender.Send(ctx, batch.toAMQPMessage(), nil)
-	}, RetryOptions(s.retryOptions))
+	}, RetryOptions(s.retryOptions), to)
 
 	return internal.TransformError(err)
```

Comment:
we need to make sure we record error on the span.
See https://github.com/open-telemetry/semantic-conventions/blob/main/docs/messaging/messaging-spans.md#span-status for the details

Reply:
TODO: we should set the `errors.type` attribute in  the [azcore/runtime](https://github.com/Azure/azure-sdk-for-go/blob/799c8579d2d8f32d3f30457cd6da736be08d9c6b/sdk/azcore/runtime/policy_http_trace.go#L147) package